### PR TITLE
JSON schema improvements (task #8712)

### PR DESCRIPTION
**Pull request by asaliev**
- Switch from obsolete `league/jsonguard` library to `justinrainbow/json-schema`.
- Introduce a json linting  library, such as `seld/jsonlint`.
- Cleanup JSON Schema files for stricter types. 
- Split and rewrite the underlying `Parser` implementations into `Parser` and `Schema` classes.
- `ModuleConfig` can now locate the schema via `$this->configType`.

### CODE REVIEW

File: src/ModuleConfig/Parser/JsonReaderTrait.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * Returns the json encoded data and applies linting if necessary.
+     *
+     * @param string $data JSON string.
+     * @param bool $lint True to apply json linting.
+     * @throws \Seld\JsonLint\ParsingException When linting is enabled and it fails.
+     * @throws \InvalidArgumentException When data is not a valid JSON object and linting is not enabled.
+     * @return \stdClass JSON object.
+     */
+    protected function dataToJson(string $data, bool $lint = false): stdClass
```
Comment (mamchenkov) [inline src/ModuleConfig/Parser/JsonReaderTrait.php]: @asaliev I think it's better to make this available as static method in `src/Utility/Convert.php`, where we already have a few conversion helpers.

Comment (asaliev) [reply src/ModuleConfig/Parser/JsonReaderTrait.php]: This trait is mostly for linting, maybe a bad choice of method name?

File: config/module_config.php [hunk 2]
```diff
@@ -25,40 +19,44 @@
 
 return [
     'ModuleConfig' => [
-        'classMapVersion' => 'V2',
+        'schemaPath' => implode(DIRECTORY_SEPARATOR, [
+            dirname(__DIR__),
```
[the remark this answers is not in the document]
Comment (asaliev) [reply config/module_config.php]: Fixed

File: src/ModuleConfig/Parser/JsonReaderTrait.php (hunk 1 above)
Comment (asaliev) [reply src/ModuleConfig/Parser/JsonReaderTrait.php]: fixed

File: config/module_config.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (asaliev) [reply config/module_config.php]: Reverted, as per our conversation. This causes issues in tests in packages which require utils.

**Pull request merged**